tools/deviceGen/input/device_limits.json

```diff
@@ -0,0 +1,2347 @@
+{
+    "constexprs": [
+        {
+            "type": "uint32_t",
+            "name": "MinMaxImageDimension2D",
+            "value": 16384
+        },
+        {
+            "type": "uint32_t",
+            "name": "MinMaxSSBOSize",
+            "value": 1073741820
+        },
+        {
+            "type": "uint16_t",
+            "name": "MaxMaxPushConstantsSize",
+            "value": 256
+        },
+        {
+            "type": "uint32_t",
+            "name": "MinMaxWorkgroupCount",
+            "value": 65535
+        },
+        {
+            "type": "uint32_t",
+            "name": "MinMaxWorkgroupInvocations",
+            "value": 256
+        },
+        {
+            "type": "int32_t",
+            "name": "MinSubPixelInterpolationOffsetBits",
+            "value": 4
+        },
+        {
+            "type": "core::bitflag<asset::IImage::E_SAMPLE_COUNT_FLAGS>",
+            "name": "NoMSor4Samples",
+            "value": "asset::IImage::E_SAMPLE_COUNT_FLAGS::ESCF_1_BIT|asset::IImage::E_SAMPLE_COUNT_FLAGS::ESCF_5_BIT",
+            "expose": false
+        },
+        {
+            "type": "uint8_t",
+            "name": "MinMaxColorAttachments",
+            "value": 8,
+            "comment": "ROADMAP 2024 and wide reports"
+        },
+        {
+            "type": "uint32_t",
+            "name": "MinInlineUniformBlockSize",
+            "value": 256,
+            "expose": false
+        }
+    ],
+    "vulkan10core": [
+        {
+            "groupComment": [
+                "Vulkan 1.0 Core"
+            ]
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxImageDimension1D",
+            "value": "MinMaxImageDimension2D"
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxImageDimension2D",
+            "value": "MinMaxImageDimension2D"
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxImageDimension3D",
+            "value": 2048
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxImageDimensionCube",
+            "value": "MinMaxImageDimension2D"
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxImageArrayLayers",
+            "value": 2048
+        },
+        {
+            "type": "uint32_t",
+            "name": "maxBufferViewTexels",
+            "value": 33554432
```

Comment:
eh I'd prefer expressions like 1<<25 which I had before, easier to ascertain what Power of Two they are